### PR TITLE
fallback for when pattern filtering filters all patterns

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
**Description:**

The filtering was sometimes filtering too much. This PR makes sure that if there was too much filtering, we don't filter at all :)

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

### CODE REVIEW

File: lib/actions/apiV2.js [hunk 1]
```diff
@@ -682,7 +682,12 @@ export const findRoute = (params) =>
                 })
               })
 
-            filteredPatterns.forEach((pattern) => {
+            // Fallback for if the filtering removes all patterns
+            // If this happens, it is not possible to know which pattern to keep
+            ;(filteredPatterns.length > 0
```
Comment (philip-cline) [inline lib/actions/apiV2.js:687]: Is the semi colon intended?

**Review by philip-cline — APPROVED** [review]
Small formatting question, but otherwise looks like a good fix

File: lib/actions/apiV2.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply lib/actions/apiV2.js:687]: Unfortunately yes this is the one time the looney goose C non compliant semicolon free style we love these days shows its issues

Comment (philip-cline) [reply lib/actions/apiV2.js:687]: Did you mean *Loosey Goose* ?

Comment (philip-cline) [reply lib/actions/apiV2.js:687]: Could the semi colon go on the line previous?

Comment (miles-grant-ibigroup) [reply lib/actions/apiV2.js:687]: Prettier put it here

**Review by amy-corson-ibigroup — COMMENTED** [review]
Wait wouldn't it be more efficient to fix `filteredPatterns` so it never returns an empty array? For example, it shouldn't ever filter out the longest pattern, right?

**Comment by miles-grant-ibigroup** [comment]
> Wait wouldn't it be more efficient to fix `filteredPatterns` so it never returns an empty array? For example, it shouldn't ever filter out the longest pattern, right?

You're right, but the issues we're seeing are only showing up on malformed GTFS, and this feels like a better approach to dealing with that than to work with it

**Review by amy-corson-ibigroup — APPROVED** [review]

**Pull request merged**